source/initial_temperature/anelasticity_temperature.cc

```diff
@@ -0,0 +1,421 @@
+        {
+          // calculate density using original parameters
+          density=original_density*(1-(original_thermal_expansivity*((temperature-273)-600))+((pressure*1e-9)/original_bulk_modulus));
+        }
+      else
+        {
+          // create fdV function to use in Brent minimization and calculate isothermal_volume_change and density using
+          // expressions in Grose & Afonso 2013
+          typedef std::pair<double, double> Result2;
+          auto vfunc = [ &,this] (double x)
+          {
+            return fdV(x, bulk_modulus, bulk_modulus_pressure_derivative, pressure);
+          };
+          Result2 r2 = boost::math::tools::brent_find_minima(vfunc,a,b,16);
+          isothermal_volume_change=r2.first;
+          compressibility=isothermal_volume_change*std::exp((gruneisen_parameter+1)*(std::pow(isothermal_volume_change,-1)-1));
+          pressure_dependent_density=reference_density*isothermal_volume_change;
+          integrated_thermal_expansivity=(2.832e-5*(temperature-273))+((0.758e-8/2)*(std::pow(temperature,2)-std::pow(273,2)));
+          density=pressure_dependent_density*(1-(compressibility*integrated_thermal_expansivity));
+        }
+      // determine J1 term (real part of complex compliance)
+      storage_compliance=unrelaxed_compliance*(1+((background_amplitude*std::pow(normalised_period,background_slope))
+                                                  /background_slope)+((std::sqrt(2*M_PI)/2)*peak_amplitude*peak_width*(1-
+                                                                      std::erf((std::log(peak_period/normalised_period))/(std::sqrt(2)*peak_width)))));
+      // determine J2 term (imaginary part of complex compliance)
+      loss_compliance=unrelaxed_compliance*(M_PI/2)*(background_amplitude*(std::pow(normalised_period,background_slope))+\
+                                                     (peak_amplitude*std::exp(-1*(std::pow(std::log(peak_period/normalised_period),2)
+                                                                                  /(2*std::pow(peak_width,2))))))+(unrelaxed_compliance*normalised_period);
+      // calculate anharmonic Vs
+      anharmonic_Vs=1/(std::sqrt(density*unrelaxed_compliance)*1e3);
+      // calculate Vs
+      anelastic_Vs=1/(std::sqrt(density*storage_compliance)*1e3);
+      // calculate attenuation
+      attenuation=loss_compliance/storage_compliance;
+      return anelastic_Vs;
+    }
+
+    template <int dim>
+    void
+    AnelasticVs2T<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.declare_entry ("Remove temperature heterogeneity down to specified depth",
+                           boost::lexical_cast<std::string>(-std::numeric_limits<double>::max()),
+                           Patterns::Double (),
+                           "This will set the heterogeneity prescribed by S20RTS or S40RTS to zero "
```

Comment:
S20RTS / S40RTS is not longer accurate, is it?

Reply:
Nope, changed!